pkg/vectorize/oct/oct.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oct
+
+import (
+	"strconv"
+
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"golang.org/x/exp/constraints"
+)
+
+var (
+	OctUint8  func([]uint8, *types.Bytes) *types.Bytes
+	OctUint16 func([]uint16, *types.Bytes) *types.Bytes
+	OctUint32 func([]uint32, *types.Bytes) *types.Bytes
+	OctUint64 func([]uint64, *types.Bytes) *types.Bytes
+	OctInt8   func([]int8, *types.Bytes) *types.Bytes
+	OctInt16  func([]int16, *types.Bytes) *types.Bytes
+	OctInt32  func([]int32, *types.Bytes) *types.Bytes
+	OctInt64  func([]int64, *types.Bytes) *types.Bytes
+)
+
+func init() {
+	OctUint8 = oct[uint8]
+	OctUint16 = oct[uint16]
+	OctUint32 = oct[uint32]
+	OctUint64 = oct[uint64]
+	OctInt8 = oct[int8]
+	OctInt16 = oct[int16]
+	OctInt32 = oct[int32]
+	OctInt64 = oct[int64]
+}
+
+func oct[T constraints.Unsigned | constraints.Signed](xs []T, rs *types.Bytes) *types.Bytes {
+	var cursor uint32
+
+	for idx := range xs {
+		octbytes := []byte(strconv.FormatUint(uint64(xs[idx]), 8))
```

Comment:
I stated a few reasons why we may prefer using bit operation in lieu of the strconv.FormatUint function in the "bin" pull request, care to check it out

Reply:
Okay, I would to see it and make changes.

Reply:
fix it